src/notifications/NotificationsHandler.tsx

```diff
@@ -0,0 +1,331 @@
+  const syncedStateRef = useRef({
+    wallets,
+    contacts,
+  });
+  const prevWalletReady = usePrevious(walletReady);
+  const onTokenRefreshListener = useRef<Callback>();
+  const foregroundNotificationListener = useRef<Callback>();
+  const notificationOpenedListener = useRef<Callback>();
+  const appStateListener = useRef<NativeEventSubscription>();
+  const appState = useRef<AppStateStatus>(null);
+  const notifeeForegroundEventListener = useRef<Callback>();
+
+  /*
+  We need to save some properties to a ref in order to have an up-to-date value
+  inside the event listener callbacks closure
+   */
+  syncedStateRef.current.wallets = wallets;
+  syncedStateRef.current.contacts = contacts;
+
+  useEffect(() => {
+    setupAndroidChannels();
+    saveFCMToken();
+    onTokenRefreshListener.current = registerTokenRefreshListener();
+    foregroundNotificationListener.current = messaging().onMessage(
+      onForegroundRemoteNotification
+    );
+    messaging().setBackgroundMessageHandler(onBackgroundRemoteNotification);
+    messaging().getInitialNotification().then(handleAppOpenedWithNotification);
+    notificationOpenedListener.current = messaging().onNotificationOpenedApp(
+      handleAppOpenedWithNotification
+    );
+    appStateListener.current = ApplicationState.addEventListener(
+      'change',
+      nextAppState => {
+        if (appState.current === 'background' && nextAppState === 'active') {
+          handleDeferredNotificationIfNeeded();
+        }
+      }
+    );
+    notifeeForegroundEventListener.current = notifee.onForegroundEvent(
+      handleNotificationPressed
+    );
+
+    return () => {
+      onTokenRefreshListener.current?.();
+      foregroundNotificationListener.current?.();
+      notificationOpenedListener.current?.();
+      appStateListener.current?.remove();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (prevWalletReady !== walletReady) {
+      handleDeferredNotificationIfNeeded();
+    }
+  }, [walletReady]);
+
+  const onForegroundRemoteNotification = (
+    remoteMessage: FirebaseMessagingTypes.RemoteMessage
+  ) => {
+    const type = remoteMessage?.data?.type;
+    if (type === NotificationTypes.walletConnect) {
+      handleWalletConnectNotification(remoteMessage);
+    } else if (remoteMessage?.notification !== undefined) {
+      handleShowingForegroundNotification(remoteMessage as FixedRemoteMessage);
+    }
+  };
+
+  const onBackgroundRemoteNotification = async (
+    remoteMessage: FirebaseMessagingTypes.RemoteMessage
+  ) => {
+    const type = remoteMessage?.data?.type;
+    if (type === NotificationTypes.walletConnect) {
+      handleWalletConnectNotification(remoteMessage);
+    }
+  };
+
+  const handleWalletConnectNotification = (
+    remoteMessage: FirebaseMessagingTypes.RemoteMessage
+  ) => {
+    const topic = remoteMessage?.data?.topic;
+
+    setTimeout(() => {
+      const requests = dispatch(requestsForTopic(topic));
```

Comment:
`Argument of type 'string | undefined' is not assignable to parameter of type 'string'.`

Reply:
just a type, WC working as expected